Label styles for design manual updates

cfarm opened the pull request:
New label styles for updated form standards in Design Manual, see https://github.com/cfpb/design-manual/issues/346
## Additions
- imports cf-typography so we can use .micro-copy class as mixin
- form label helper text class created for instructional label text
- form label error class added for error messages on field-level validation errors
## Changes
- form labels now use h4 instead of h5 style
## Testing
- check em out here http://cfarm.github.io/design-manual/ui-toolkit/form-fields.html
## Review
- @Scotchester 
- @sonnakim 
## Screenshots

![screen shot 2015-09-24 at 8 39 08 pm](https://cloud.githubusercontent.com/assets/702526/10090352/8bc40a46-62fc-11e5-9c4f-793e2e7956eb.png)
![screen shot 2015-09-24 at 8 39 18 pm](https://cloud.githubusercontent.com/assets/702526/10090357/8f9e437a-62fc-11e5-879c-0643dee1671e.png)
## Notes

Related PRs
- fieldset legend added in [cf-core PR 47](https://github.com/cfpb/cf-core/pull/47)
- variable overrides in:
  - capital-framework https://github.com/cfpb/capital-framework/pull/185
  - generator-cf https://github.com/cfpb/generator-cf/pull/96
## Checklist
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [ ] Passes all existing automated tests
- [ ] New functions include new tests
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged
- [ ] Visually tested in supported browsers and devices 
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

cfarm committed:
update form label to h4 style

cfarm committed:
import typography .micro-copy, add helper text label

cfarm committed:
rm helper text margin

cfarm committed:
margin bottom for help label

cfarm committed:
margin bottom

cfarm committed:
error label and icon

cfarm committed:
rm old icon styles

cfarm committed:
Fix missing ampersand

cfarm committed:
remove old icon positioning

jimmynotjim commented:
Is the legend shown in the screenshot not included in this PR?

jimmynotjim commented:
NVM, found the legend text changes here -> https://github.com/cfpb/cf-core/pull/47

:+1: to this PR

Scotchester commented on src/cf-forms.less:
```diff
@@ -56,15 +60,32 @@
 
 .form-label-header {
-    .h5();
+    .h4();
```
This change should be happening in cf-core's `label` styling. `.form-label-header` is a separate pattern for when you need a heading above the form label, e.g. "Date range" would be the `.form-label-header` above two fields labeled "From" and "To".

cfarm replied:
@sonnakim can you clarify which element the new standard affects in this case? Just the default label style or the other case Scott mentioned, too?

Scotchester replied:
Bump for @sonnakim feedback.

Scotchester commented on src/cf-forms.less:
```diff
@@ -56,15 +60,32 @@
+.form-label-helper-text {
+    .micro-copy();
+    color: @label-helper;
+    margin-bottom: unit(10px / 14px, em);
```
Perhaps note in a comment that the `14px` comes from the font size specified in `.micro-copy()`.

Scotchester commented on src/cf-forms.less:
```diff
@@ -56,15 +60,32 @@
+
+.form-label-error {
+    .error + & {
+      margin-top: unit(15px / 16px, em);      
```
- inconsistent indentation
- Please use `@base-font-size-px` instead of `16px`
- Let's unnest this into a single `.error + .form-label-error` declaration until such time as nesting would be beneficial.

Scotchester commented on src/cf-forms.less:
```diff
@@ -197,25 +218,13 @@ textarea {
-.success + .@{cf-forms_input-icon-class} {
-    color: @input-success;
-}
-
```
Per the note from @sonnakim, please restore these removals.

Scotchester commented:
Please run the full `grunt` command to generate docs and demo updates.

cfarm committed:
restore error/success/warning, comments + style tweaks

jimmynotjim commented on src/cf-forms.less:
```diff
@@ -56,15 +60,31 @@
+.form-label-helper-text {
+    .micro-copy();
+    color: @label-helper;
+    // 14px font-size value is from .micro-copy
```
@Scotchester should we update `.micro-copy()` to expose a `@font-size` variable like the headings to avoid needing to leave notes like this?

Scotchester replied:
Yeah, I think that would be a good idea.

jimmynotjim replied:
Done https://github.com/cfpb/cf-typography/pull/41

jimmynotjim commented on src/cf-forms.less:
```diff
@@ -197,26 +217,30 @@ textarea {
+    color: @input-error;
+}
+
+// Deprecated
```
Anyone know if you can send warnings to the console with Less like you can in Sass with [@warn](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#_6)?

wpears replied:
Yeah, you can hack it by doing something like:

```
// Deprecated
 .warning + .@{cf-forms_input-icon-class} {
     color: `(function(){
       console.log('cf-forms warning is deprecated!');
       return @{input-warning};
     })()`;
 }
```

wpears replied:
Could also do something like:

```
@warn: `function(msg){console.log(msg ? msg : 'Deprecated');}`;

.warning + .@{cf-forms_input-icon-class}{
    color: `@{warn}(), @{input-warning}`
}
```

jimmynotjim replied:
Wait, you can write JS in your Less? I had no idea that was possible.

Scotchester commented:
Heads up for later: Merge conflicts now.

cfarm committed:
Merge branch 'gh-pages' into labels

Conflicts:
	src/cf-forms.less

cfarm committed:
add cf-typography bower dependency

cfarm committed:
compiled demo + docs

cfarm committed:
add deprecated warning for input icons

cfarm committed:
less hax pt ii

cfarm committed:
more js

cfarm committed:
let's try this way

cfarm committed:
no more js

cfarm committed:
deprecated icon class warnings in console

cfarm committed:
demo and docs

cfarm commented:
updated demo + docs, still waiting on @sonnakim guidance on the form-label-header element (see comments above)

I added the recommended deprecation console warnings, too:

<img width="620" alt="screen shot 2015-10-09 at 2 16 37 pm" src="https://cloud.githubusercontent.com/assets/702526/10462868/0b11ca20-71af-11e5-88be-04e548772fb6.png">

Scotchester commented:
Obsolete.
